test/WebJobs.Script.Tests.Integration/CosmosDBTrigger/CosmosDBTriggerEndToEndTestsBase.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Azure.Documents;
+using Microsoft.Azure.Documents.Client;
+using Microsoft.Azure.WebJobs.Host;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests.CosmosDBTrigger
+{
+    public abstract class CosmosDBTriggerEndToEndTestsBase<TTestFixture> :
+        EndToEndTestsBase<TTestFixture> where TTestFixture : CosmosDBTriggerTestFixture, new()
+    {
+        public CosmosDBTriggerEndToEndTestsBase(TTestFixture fixture) : base(fixture)
+        {
+        }
+
+        protected async Task CosmosDBTriggerToBlobTest()
+        {
+            // CosmosDB tests need the following environment vars:
+            // "AzureWebJobsCosmosDBConnectionString" -- the connection string to the account
+
+            // Waiting for the Processor to acquire leases
+            await Task.Delay(10000);
```

Comment:
Is there a better way? This will be running in CI, so a 10 sec. wait is a bit painful

Reply:
@ealsur -- Did you ever investigate this hard-coded delay in the e2e tests? Do we get any log or notification when the listener is ready?

Reply:
BTW -- this already exists in v1.x (it was a copy/paste) so I'm not going to block on it. If there's a fix, we should do it in both places.